documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -834,6 +831,103 @@ public interface IJoystickInputHandler : IInputHandler
 
 `HandleHatMove` must be called when any value of `JoystickState.Hats` changes.
 
+# Input Maps
+
+## Presentation
+Input maps are designed to be easyer to the user to handle actions with more than one input.
+At first view it can looks useless and a bad practice, but this is complemtelly useful for cross-plataform programs, being games an example.
+
+This is an pseudocode example of what the Input Map system solves:
+
+Without inputMaps:
+```cs
+inputContext.Update();
+
+private Dictionary<string, bool> pressedKeys = new();
+
+inputContext.Gamepads.ThumbstickMove += @event => {Move(@event.Value)};
+inputContext.Joysticks.AxisMove += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+With inputMaps:
+```cs
+inputContext.Update();
+
+// This single line get rid of the two in the example above. TWO ISN'T THE LIMIT!
+inputContext.Map.GetAction("Move").Axis += @event => {Move(@event.Value)};
+
+private Move(Vector2D<float> value)
+{
+    // Code for movement here
+}
+```
+
+## Implementation
+Each InputContext have one reference to an InputMap.
+This is how the InputMap class work:
+```cs
+public partial class InputMap
+{
+    public InputMapAction GetAction(string name);
+    public void AddAction(InputMapAction action, string name);
+    public void RemoveAction(string name);
+    public static InputMap Load(string path);
+    public static void Save(InputMap source, string path);
+    public static string Compile(InputMap source);
+}
+```
+
+`GetAction` method is the main entry point of the system. It allows the use to get an mapped action by it configurated name.
+
+```cs
+public partial class InputMapAction
+{
+    public bool IsPressed { get; }
```

Comment:
What does this mean for actions that aren't boolean on/off actions?

Reply:
Good question... I was thinking about creating classes for each type of input, you think it's better in that way? As it's a class there's no reason to put all possible returns in the same type

Reply:
I do not think it's better in that way. I don't know what could make it better to be honest.

Reply:
Tbh i really do not want to do it with separate classes also... I will try to think a way to everything works just inside the InputMap class, without unnecessary data access